plumbum/machines/remote.py

```diff
@@ -173,6 +173,7 @@ def __init__(self, encoding="utf8", connect_timeout=10, new_session=False):
         self.uname = self._get_uname()
         self.env = RemoteEnv(self)
         self._python = None
+        self._program_cache = {}
```

Comment:
I thought through ways to be more clever here with functools.lru_cache, but they end up being worse (it caches all arguments, including `self`). So I think this is good.